### PR TITLE
Handle failures when operating with symptom logs

### PR DESCRIPTION
**Pull request by aledustet**
Why:
----

We need to be able to recover from errors operating on symptom logs or check ins.

This Commit:
----
- Handle failures on log context operations
- Show flash error message when adding symptoms fails
- Show flash error message when deleting log symptoms fails
- Show flash error message when updating log symptoms fails

### CODE REVIEW

File: src/MyHealth/SymptomLogContext.tsx [hunk 1]
```diff
@@ -84,31 +89,56 @@ export const SymptomLogProvider: FunctionComponent = ({ children }) => {
...
 
   const updateLogEntry = async (updatedLogEntry: SymptomLogEntry) => {
-    await modifyLogEntry(updatedLogEntry)
-    await fetchLogEntries()
+    try {
+      await modifyLogEntry(updatedLogEntry)
+      await fetchLogEntries()
+      return SUCCESS_RESPONSE
+    } catch (e) {
+      return failureResponse(e.message)
```
Comment (johnschoeman) [inline src/MyHealth/SymptomLogContext.tsx:117]: What do you think about added `Logger.error()`'s for these operations?  Maybe it makes sense to do this in the native module?

**Review by johnschoeman — APPROVED** [review]
Nice!

File: src/MyHealth/SymptomLogContext.tsx (hunk 1 above)
Comment (aledustet) [reply src/MyHealth/SymptomLogContext.tsx:117]: I believe we are logging the errors on the `failureResponse` method. Would it be better to move it higher to the `nativeModule`?

Comment (johnschoeman) [reply src/MyHealth/SymptomLogContext.tsx:117]: ah, no, just misread the code. imo, its fine where it is at.

File: src/MyHealth/Today.spec.tsx [hunk 2]
```diff
@@ -55,6 +57,34 @@ describe("Today", () => {
...
+          <Today />
+        </SymptomLogContext.Provider>,
+      )
+
+      fireEvent.press(getByLabelText("Good"))
+
+      await waitFor(() => {
+        expect(showMessageSpy).toHaveBeenCalledWith(
+          expect.objectContaining({
+            message: "Sorry, we could not add todays check in",
```
Comment (devinjameson) [inline src/MyHealth/Today.spec.tsx:82]: @aledustet Does this need to be translated? Also I think it wants to be `add today's check in` with an apostrophe in `today`.

**Pull request merged**